performance/src/main/kotlin/org/jetbrains/ring/launcher.kt

```diff
@@ -24,12 +24,36 @@ val BENCHMARK_SIZE = 100
 
 //-----------------------------------------------------------------------------//
 
-class Launcher(val numWarmIterations: Int, val numberOfAttempts: Int) {
+class Launcher(val numWarmIterations: Int, val numberOfAttempts: Int, val jsonReport: String?) {
     class Results(val mean: Double, val variance: Double)
 
+    enum class Status(val value: String) {
+        PASSED("PASSED"),
+        FAILED("FAILED")
+    }
+
+    data class BenchmarkResult(val name: String, val status: Status,
+                          val score: Double, val runtimeInUs: Double,
+                          val repeat: Int, val warmup: Int) {
+        override fun toString(): String {
+            return """
+        {
+            "name": "$name",
+            "status": "${status.value}",
+            "score": ${score.toString()},
+            "runtimeInUs": ${runtimeInUs.toString()},
+            "repeat": ${repeat.toString()},
+            "warmup": ${warmup.toString()}
+        }
+        """
+
+        }
+    }
+
     val results = mutableMapOf<String, Results>()
+    val bechmarkResults = mutableListOf<BenchmarkResult>()
 
-    fun launch(benchmark: () -> Any?): Results {                          // If benchmark runs too long - use coeff to speed it up.
+    fun launch(benchmark: () -> Any?, name: String): Unit {                          // If benchmark runs too long - use coeff to speed it up.
```

Comment:
It's better to move writing to JSON out from `Launcher` to separate class because the only responsibility of `Launcher` is benchmark running.